### ISSUE TITLE
FEATURE FREEZE?

### ISSUE THREAD

**Issue by rhyolight** [nupic-community/nupic.visualizations#49]
@breznak @jefffohl I think this project is very cool, and I'm glad you are working on it. I'm ready to use it if it in more than one project, including River View. 

**_But_**, at this point I think you should focus on getting this codebase ready for production and stop all work on new features until it is actually deployed and being used in a few places. We are going to find bugs and new issues as we attempt to deploy this charting tech, and it s better to find out any **big** issues now before any further architecture changes are made that might somehow affect deployment strategies. (I am talking about issues like #42, #43 & #48.)

What do you guys think?

**Comment by jefffohl** [comment]
Sure. If it is getting a lot of interest, then I suppose we should do releases. We can target issues to planned releases.

Speaking of deployment - if modularization is a big priority - that is, make it so that any number of nupic.visualizations can be dropped on a page with a minimum of configuration - then we should make that a priority issue. It isn't really designed that way right now. Right now, it is designed to be an entire stand-alone application, as opposed to being a part of some other app.

**Comment by rhyolight** [comment]
Yes, and I much prefer it to be a drop-in charting tool. I think it can be
both if we treat this repo as a client-side drop-in charting tool and add
another server repo that makes is stand-alone.

---

Matt Taylor
OS Community Flag-Bearer
Numenta

**Comment by breznak** [comment]
To me the feature freeze is a bit confusing - depends what you want to use this for. For me, for example, the code is OK enough for a `1.0`. #40 is not a high prio for me, I'd like to see support for the streaming functionality #17 and anomalies ( #3 #31 )

**Comment by jefffohl** [comment]
Since @rhyolight needs streaming as well, perhaps we should make that the highest priority?

**Comment by rhyolight** [comment]
My concern when creating this issue was simply that too much up-front engineering would be done before we had the charting system actually being used somewhere. I think I had the wrong impression that the end goal was a drop-in JavaScript library that could be used for live-streaming temporal data. 

If you guys are willing to do the work now to ensure I can use this charting system in my own web applications, I would be happy. Otherwise, I'll continue to use Dygraphs directly.

**Comment by jefffohl** [comment]
I want to make it something you can use Matt. We just need to define the interface.

**Comment by breznak** [comment]
@rhyolight what's the missing parts for you? The #40 and #17 ?

**Comment by jefffohl** [comment]
I think we also need to know what to leave out. What would be the responsibility of the charting module, and what would be the responsibility of the parent app? What UI elements do we include, etc.

**Comment by rhyolight** [comment]
All I need right now is #40. Streaming can be done later.

**Comment by jefffohl** [comment]
Can we chat on Gitter? I need more detail.

**Comment by rhyolight** [comment]
I think the UI component would be this: 

![screen shot 2015-12-14 at 5 14 02 pm](https://cloud.githubusercontent.com/assets/15566/11799124/231f5a3a-a286-11e5-8b4c-44a85aef05dc.png)

**Comment by breznak** [comment]
I would add the file selection to the UI, actually the existing app minus the header... 
Make it into a "namespace secure" div and the app can be easily dropped anywhere, have multiple occurrences at a page, etc. 
I'd hope to add more controls to the UI in a near future, but it'll be in the div, thus transparent update to any apps using it
